### PR TITLE
new function uninstall, forwards to remove.packages

### PR DESCRIPTION
**Pull request by krlmlr**
Use case: Easy uninstall for one-off packages that are installed for testing purposes.

Closes #817 (=includes it).

### CODE REVIEW

File: R/uninstall.r [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#' @param ... additional arguments passed to \code{\link{remove.packages}}.
+#' @export
+#' @family package installation
+#' @seealso \code{\link{with_debug}} to install packages with debugging flags
+#'   set.
+uninstall <- function(pkg = ".", unload = TRUE, quiet = FALSE, ...) {
+
+  pkg <- as.package(pkg)
+
+  if (unload && pkg$package %in% loaded_packages()$package) unload(pkg)
```
Comment (hadley) [inline R/uninstall.r]: Can you put `unload()` on a new line please? (And similarly below)

**Comment by hadley** [comment]
Looks good - could you add a bullet to NEWS (and maybe a unit test?)

**Comment by krlmlr** [comment]
Done. Contains two unrelated code quality improvements in install.r and test-dll.r, hope it's ok.

File: tests/testthat/test-uninstall.r [hunk 2]
```diff
@@ -0,0 +1,23 @@
+context("Uninstall")
+
+test_that("uninstall() unloads and removes from library", {
+
+  # Make a temp lib directory to install test package into
+  old_libpaths <- .libPaths()
```
Comment (hadley) [inline tests/testthat/test-uninstall.r]: I wonder if it would be useful to pull this out into `with_temp_lib` or something?

Comment (krlmlr) [inline tests/testthat/test-uninstall.r]: Agreed, but let's not clutter this pull request even more.

**Comment by hadley** [comment]
Thanks!

**Pull request merged**